cody-diff.el

```diff
@@ -0,0 +1,116 @@
+;;; cody-diff.el -- Provides Myers diff functionality -*- lexical-binding: t; -*-
+;; Copyright (C) 2024 Sourcegraph, Inc.
+
+;; Version: 0.1
+;; Author: Steve Yegge <steve.yegge@gmail.com>
+;; URL: https://github.com/sourcegraph/emacs-cody
+
+;;; Commentary:
+;; This provides a very quick-and-dirty list differ.  It is intended
+;; only for diffing within relatively short single sequences of under a
+;; few hundred characters, for code autocompletion use cases.
+
+;;; Code:
+
+(eval-when-compile (require 'cl-lib))
+
+(defvar cody--list-diff-cache (make-hash-table :test 'equal)
```

Comment:
Since you're peeling off the front of the same lists, could you use `'eq` here for a much faster test?